### PR TITLE
Add coverage for abort/error during an in-flight close

### PR DESCRIPTION
**Pull request by domenic**
Closes https://github.com/whatwg/streams/issues/693.

### CODE REVIEW

**Comment by w3c-bots** [comment]
These tests are now available on [w3c-test.org](https://w3c-test.org/submissions/5241/)

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/215887840)

# Firefox (nightly channel) #
Testing web-platform-tests at revision a00fff852008940791bd55b1eaff712be090f07e
Using browser at version BuildID 20170327110256; SourceStamp 9577ddeaafd85554c2a855f385a87472a089d5c0
Starting 10 test iterations

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/215887841)

# Chrome (unstable channel) #
Testing web-platform-tests at revision a00fff852008940791bd55b1eaff712be090f07e
Using browser at version 59.0.3047.0 dev
Starting 10 test iterations

File: streams/writable-streams/aborting.js [hunk 1]
```diff
@@ -829,6 +924,87 @@ promise_test(t => {
...
+
+    return Promise.all([
+      promise_rejects(t, error2, writer.ready,
+                      'writer.ready must reject with the error passed to the controller\'s error method'),
+      flushAsyncEvents()
+    ]);
+  }).then(() => {
+    assert_array_equals(
+        events, ['abortPromise'],
+        'writePromise and writer.closed must not be fulfilled/rejected yet even after writer.abort()');
```
Comment (tyoshino) [inline streams/writable-streams/aborting.js]: s/writePromise/closePromise/

File: streams/writable-streams/aborting.js [hunk 2]
```diff
@@ -829,6 +924,87 @@ promise_test(t => {
...
+      promise_rejects(t, error2, writer.closed,
+                      'writer.closed must reject with the error passed to the controller\'s error method'),
+      flushAsyncEvents()
+    ]);
+  }).then(() => {
+    assert_array_equals(events, ['abortPromise', 'closePromise', 'closed'],
+                        'abortPromise, closePromise and writer.closed must fulfill/reject');
+
+    return Promise.all([
+      closePromise,
```
Comment (tyoshino) [inline streams/writable-streams/aborting.js]: this can be included in the Promise.all at L982?

File: streams/writable-streams/aborting.js [hunk 3]
```diff
@@ -829,6 +924,87 @@ promise_test(t => {
...
+      promise_rejects(t, error2, writer.ready,
+                      'writer.ready must be still rejected with the error passed to the controller\'s error method')
```
Comment (tyoshino) [inline streams/writable-streams/aborting.js]: maybe this too as we do nothing in this block?

**Review by tyoshino — APPROVED** [review]
lgtm

**Comment by domenic** [comment]
@tyoshino if you could take another look, I believe I addressed all your comments in this review and others. I'd like to be sure before merging though.

**Review by tyoshino — APPROVED** [review]
lgtm!

**Pull request merged**